### PR TITLE
WT-9249 Compare and fix the start durable timestamp if it is greater than OOO timestamp (v5.0 backport)

### PR DESCRIPTION
**Pull request by Sean04**
Performing a prepared insert rollback can lead to a situation where
the datastore and history store may have the same update and inserting it
back again to the history store triggers the OOO timestamp handling.

Compare and fix the existing history store entry start durable timestamp
also to avoid generating the condition where the start durable timestamp
is greater than the stop timestamp.

(cherry picked from commit https://github.com/wiredtiger/wiredtiger/commit/462db2dbb2ed77b516dd35addf5dce4f4978f3c1)

Co-authored-by: Hari Babu Kommi <haribabu.kommi@mongodb.com>
(cherry picked from commit https://github.com/wiredtiger/wiredtiger/commit/ba455f3c9bd7c3fd1d2b43fb659c53f9822c0c6b)

### CODE REVIEW

File: src/cursor/cur_hs.c [hunk 1]
```diff
@@ -1047,6 +1047,9 @@ __curhs_update(WT_CURSOR *cursor)
     hs_upd->durable_ts = hs_cursor->time_window.durable_start_ts;
     hs_upd->txnid = hs_cursor->time_window.start_txn;
 
+    WT_ASSERT(session,
```
Comment (Sean04) [inline src/cursor/cur_hs.c:1050]: Do we want to remove this assert too?

Comment (kommiharibabu) [reply src/cursor/cur_hs.c:1050]: No need to remove this assertion. This API is called only from prepare commit/rollback. In those scenarios always the durable timestamp will be larger. While going through these changes, It is better to backport the WT-8425 into 5.0 as it fixes a history store bug.

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Review by kommiharibabu — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**